modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/runtime/execution/SdkInternalContext.java

```diff
@@ -56,10 +60,12 @@ public static SdkInternalContext from(CoreEvent event) {
   private Function<Context, Context> innerChainSubscriberContextMapping = identity();
 
   public void removeContext(ComponentLocation location, String eventId) {
+    LOGGER.debug("Removing context at location - {} for event - {}", location.getLocation(), eventId);
```

Comment:
I'm worried that in some corner case here `location` could be `null`. Hence, we would be causing an NPE that wouldn't otherwise had happened.

We could use the version of this with the supplier which would make this theoretical problem never happen unless the logging level is DEBUG.

But maybe we also want to check if `location` is `null` when logging.

Maybe something like this:
```suggestion
    LOGGER.debug("Removing context at location - {} for event - {}", () -> {location != null ? location.getLocation() : "null"}, eventId);
```

Reply:
Of course this applies to all the other new log messages.

Reply:
Completed the changes suggested